lib/src/publication/local.dart

```diff
@@ -27,4 +33,39 @@ class LocalTrackPublication<T extends LocalTrack> extends TrackPublication<T> {
 
   /// Unmute the track associated with this publication
   Future<void> unmute() async => await track?.unmute();
+
+  @internal
+  void updatePublishingLayers(List<lk_rtc.SubscribedQuality> layers) async {
+    //
+    final params = track?.sender?.parameters;
+    if (params == null) return;
+
+    final encodings = params.encodings;
+    if (encodings == null) return;
+
+    bool didChange = false;
+
+    for (final encoding in encodings) {
+      final layer = layers.firstWhereOrNull((e) {
+        // If there is exact match, use it
+        if (e.quality.toRid() == encoding.rid) return true;
+        // Use first layer found if not simulcast
+        if (encoding.rid == null && encodings.length == 1) return true;
```

Comment:
when RID isn't defined, we should specifically match it to LOW quality layer. The protocol doesn't guarantee layers to come in in any specific order. (even tho low comes in first).